src/docs/framework-integration/react.md

```diff
@@ -202,8 +218,153 @@ import { MyComponent } from 'component-library-react';
 
 With that, your component is now available to be used like any other React component.
 
+## API
+
+### componentCorePackage
+
+**Optional**
+
+**Default: The `components.d.ts` file in the Stencil project's `package.json` types field**
+
+**Type: `string`**
+
+The title of the Stencil package where components are available for consumers (i.e. the `name` property value in your Stencil project's `package.json`).
+This is used during compilation to write the correct imports for components.
+
+For a starter Stencil project generated by running:
+
+```bash
+npm init stencil component my-component-lib
+```
+
+The `componentCorePackage` would be set to:
+
+```ts
+// stencil.config.ts
+
+export const config: Config = {
+  ...,
+  outputTargets: [
+    reactOutputTarget({
+      componentCorePackage: 'my-component-lib',
+      // ... additional config options
+    })
+  ]
+}
+```
+
+Which would result in an import path like:
+
+```js
+import { defineCustomElement as defineMyComponent } from 'my-component-lib/components/my-component.js';
+```
+
+> **NOTE:** Although this field is optional, it is _highly_ recommended that it always be defined to avoid potential issues with paths not being generated correctly when combining other API arguments.
+
+### customElementsDir
+
+**Optional**
+
+**Default: 'dist/components'**
+
+If [includeImportCustomElements](#includeimportcustomelements) is `true`, this option can be used to specify the directory where the generated
+custom elements live. This value only needs to be set if the `dir` field on the `dist-custom-elements` output target was set to something other than
+the default directory.
+
+### excludeComponents
+
+**Optional**
+
+**Default: `[]`**
+
+**Type: `string[]`**
+
+This lets you specify component tag names for which you don't want to generate React wrapper components. This is useful if you need to write framework-specific versions of components. For instance, in Ionic Framework, this is used for routing components - like tabs - so that
+Ionic Framework can integrate better with React's Router.
+
+### includeDefineCustomElements
+
+**Optional**
+
+**Default: `true`**
+
+**Type: `boolean`**
+
+If `true`, all Web Components will automatically be registered with the Custom Elements Registry. This can only be used when lazy loading Web Components and will not work when `includeImportCustomElements` is `true`.
+
+### includeImportCustomElements
+
+**Optional**
+
+**Default: `undefined`**
+
+**Type: `boolean`**
+
+If `true`, the output target will import the custom element instance and register it with the Custom Elements Registry when the component is imported inside of a user's app. This can only be used with the [Custom Elements Bundle](/docs/custom-elements) and will not work with lazy loaded components.
```

Comment:
I think this is just an issue with the `dist-custom-elements` and `dist-custom-elements-bundle` target terminologies being used too loosely in the past. I'll update this to not say "Bundle" but keep the link the same. However, there may be some additional cleanup we need to address for V3 with the `loaderDir` config option since that is expecting a `defineCustomElements` function that would only exist if `customElementsExportBehavior: 'bundle'` is set.

Reply:
> I'll update this to not say "Bundle" but keep the link the same. 

👍 

> However, there may be some additional cleanup we need to address for V3 with the loaderDir config option since that is expecting a defineCustomElements function that would only exist if customElementsExportBehavior: 'bundle' is set.

Can you capture those TODOs somewhere so they don't fall through the cracks?